### PR TITLE
Added metrics collection and metrics-based dashboards info to MongoDB

### PR DESCRIPTION
**Pull request by yasar-sumologic**
## Purpose of this pull request

This pull request Adds metrics collection and metrics-based dashboards info to MongoDB 

## Select the type of change

- [ ] Minor Changes - Typos, formatting, slight revisions
- [x] Update Content - Revisions, updating sections
- [ ] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

## Ticket (if applicable)

https://sumologic.atlassian.net/browse/SUMO-243921

### CODE REVIEW

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 1]
```diff
@@ -11,15 +11,17 @@ import TabItem from '@theme/TabItem';
 
 <img src={useBaseUrl('img/integrations/databases/mongodb.png')} alt="Thumbnail icon" width="120"/><img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>
 
-[MongoDB](https://www.mongodb.com/why-use-mongodb#:~:text=MongoDB%20is%20a%20document%20database,development%20teams%20using%20agile%20methodologies.) is a source-available cross-platform document-oriented database program. The Sumo Logic app for MongoDB supports logs and metrics from the open source version of MongoDB. The App is tested on the 4.4.4 version of MongoDB.
+[MongoDB](https://www.mongodb.com/why-use-mongodb#:~:text=MongoDB%20is%20a%20document%20database,development%20teams%20using%20agile%20methodologies.) is a source-available cross-platform document-oriented database program. The Sumo Logic app for MongoDB supports logs and metrics from the open source version of MongoDB.
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:14]: We can specify the version of mongo which we have tested against.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 2]
```diff
@@ -29,9 +31,16 @@ Following are the [Fields](/docs/manage/fields/) which will be created as part o
...
 - **`sumo.datasource`**. has a fixed value of **mongodb**.
+* **`db.node.name`**. Has the value of host name of the machine which is being monitored.
 
 ## Prerequisites
 
+### For metrics collection
+
+The MongoDB receiver fetches stats from a MongoDB instance using the [golang mongo driver](https://github.com/mongodb/mongo-go-driver). Stats are collected via MongoDB's `dbStats` and `serverStatus` commands. The MongoDB receiver extracts values from the result and converts them to OpenTelemetry metrics.
+
+### For logs collection
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:42]: Will need to add instruction for profiling level setting in mongo for sending logs.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 3]
```diff
@@ -79,7 +88,18 @@ import SetupColl from '../../../reuse/apps/opentelemetry/set-up-collector.md';
...
-In this step the user needs to provide the path to the mongo db log file configured as part of above steps. Typically the logs are located at the location: `/var/log/mongodb/mongodb.log`.
+OpenTelemetry works with a [configuration](https://opentelemetry.io/docs/collector/configuration/) yaml file with all the details concerning the data that needs to be collected. For example, it specifies the location of a log file that is read and sent to the Sumo Logic platform.
+
+In this step, you will configure the yaml file required for MongoDB collection.
+
+Below are the inputs required:
+
+- **`Endpoint (no default)`**. The hostname and port of the MognoDB instance, separated by a colon. (For example: `localhost:27017`.)
+- **Logs Path**. Enter the path to the log file for your MognoDB instance.
+- **username**. Enter the MognoDB username.
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:99]: We should mention that username/password is optional

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 4]
```diff
@@ -79,7 +88,18 @@ import SetupColl from '../../../reuse/apps/opentelemetry/set-up-collector.md';
...
+- **password**. Enter the MognoDB password.
+
+User needs to provide the path to the mongo db log file configured as part of above steps. Typically the logs are located at the location: `/var/log/mongodb/mongodb.log`.
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:102]: We should include this in logs PAth description itself and not mention this line seperately.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 5]
```diff
@@ -257,24 +277,54 @@ Use this dashboard to:
 
 <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/MongoDB-OpenTelemetry/MongoDB-Query-Logs.png' alt="Query Logs" />
 
+### Sharding
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:280]: Do we have a sharding dashboard ?

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 6]
```diff
@@ -257,24 +277,54 @@ Use this dashboard to:
...
+
+-   Identify Sharding errors and warnings.
+-   Gain insights into Chunk operations.
+
+<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/MongoDB-OpenTelemetry/MongoDB-Sharding.png' alt="Sharding" />
+
 ### Replication Logs
 
-The **MongoDB - Replication Logs** dashboard shows replica deletes/updates/inserts trend and replica state.
+The **MongoDB - Replication Logs** dashboard shows replication events, errors, warnings, and nodes.
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:293]: Have we made changes to this dashboard . This seem to be a log dashbaord.. and we have added replication related metric dashboard seperately. Why are we changing the description for this dashboard ?

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 2 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:42]: it is already there on line number 46

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 4 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:102]: fixed it.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 5 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:280]: Yes it is already there & it is logged based dashboard

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 6 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:293]: fixed it.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 1 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md]: fixed it

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 3 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md]: fixed it.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 2 above)
Comment (sumoanema) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:42]: That is log level. We need to add instruction for setting profiling level

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 7]
```diff
@@ -79,7 +90,18 @@ import SetupColl from '../../../reuse/apps/opentelemetry/set-up-collector.md';
...
+Below are the inputs required:
+
+- **`Endpoint (no default)`**. The hostname and port of the MognoDB instance, separated by a colon. (For example: `localhost:27017`.)
+- **`Logs Path`**. Enter the path to the log file for your MognoDB instance.
+- **`username (optional)`**. Enter the MognoDB username.
+- **`password (optional)`**. Enter the MognoDB password.
+
+User needs to provide the path to the mongo db log file configured as part of above steps. Typically the logs are located at the location: `/var/log/mongodb/mongodb.log`.
 
 You can add any custom fields which you want to tag along with the data ingested in Sumo. Click on the **Download YAML File** button to get the yaml file.
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:106]: We also need to mention if authentication for mongo is configured then they need to download env file. Please check the content in any of the other app docs.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 8]
```diff
@@ -79,7 +90,18 @@ import SetupColl from '../../../reuse/apps/opentelemetry/set-up-collector.md';
...
-In this step the user needs to provide the path to the mongo db log file configured as part of above steps. Typically the logs are located at the location: `/var/log/mongodb/mongodb.log`.
+OpenTelemetry works with a [configuration](https://opentelemetry.io/docs/collector/configuration/) yaml file with all the details concerning the data that needs to be collected. For example, it specifies the location of a log file that is read and sent to the Sumo Logic platform.
+
+In this step, you will configure the yaml file required for MongoDB collection.
+
+Below are the inputs required:
+
+- **`Endpoint (no default)`**. The hostname and port of the MognoDB instance, separated by a colon. (For example: `localhost:27017`.)
+- **`Logs Path`**. Enter the path to the log file for your MognoDB instance.
+- **`username (optional)`**. Enter the MognoDB username.
```
Comment (sumoanema) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:101]: For username and password, can we mention something on the lines of mongodb receiver doc : 
**username (optional)** : If authentication is required, the user can with **clusterMonitor** permissions can be provided here.
**password (optional)**: If authentication is required, the password can be provided here.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 5 above)
Comment (sumoanema) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:282]: So was the doc missing the description for sharding dashboard ?

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 4 above)
Comment (sumoanema) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:104]: I do not see it fixed. You need to remove this line from here and place it under "Logs Path"

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 7 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:106]: fixed it.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 8 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md]: fixed it.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 4 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:104]: fixed it.

Comment (sumoanema) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:104]: Can you remove line 104. There is no use of it after we have added details above

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 7 above)
Comment (sumoanema) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:106]: in step3 after this there should be steps to include env file under linux tab. Please refer mysql doc for it.

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 4 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:104]: Done

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md (hunk 7 above)
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:106]: Ok

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 9]
```diff
@@ -11,15 +11,17 @@ import TabItem from '@theme/TabItem';
 
 <img src={useBaseUrl('img/integrations/databases/mongodb.png')} alt="Thumbnail icon" width="120"/><img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>
 
-[MongoDB](https://www.mongodb.com/why-use-mongodb#:~:text=MongoDB%20is%20a%20document%20database,development%20teams%20using%20agile%20methodologies.) is a source-available cross-platform document-oriented database program. The Sumo Logic app for MongoDB supports logs and metrics from the open source version of MongoDB. The App is tested on the 4.4.4 version of MongoDB.
+[MongoDB](https://www.mongodb.com/why-use-mongodb#:~:text=MongoDB%20is%20a%20document%20database,development%20teams%20using%20agile%20methodologies.) is a source-available cross-platform document-oriented database program. The Sumo Logic app for MongoDB supports logs and metrics from the open source version of MongoDB.The App is tested on the 7.0.14 version of MongoDB.
```
Comment (JV0812) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:14]:
```suggestion
[MongoDB](https://www.mongodb.com/why-use-mongodb#:~:text=MongoDB%20is%20a%20document%20database,development%20teams%20using%20agile%20methodologies.) is a source-available cross-platform document-oriented database program. The Sumo Logic app for MongoDB supports logs and metrics from the open source version of MongoDB. The app is tested on the 7.0.14 version of MongoDB.
```

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 10]
```diff
@@ -11,15 +11,17 @@ import TabItem from '@theme/TabItem';
...
 
 <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/MongoDB-OpenTelemetry/MongoDB-Schematics.png' alt="Schematics" />
 
-## Log types
+## Log and metrics types
 
-The MongoDB logs are generated in files as configured in the configuration file `/var/log/mongodb/mongodb.log`. For more details on MongoDB logs, see [this](https://docs.mongodb.com/manual/reference/log-messages/) link.
+This app supports metrics and logs for MongoDB instance. The MongoDB logs are generated in files as configured in the configuration file `/var/log/mongodb/mongodb.log`. For more details on MongoDB logs, see [this](https://docs.mongodb.com/manual/reference/log-messages/) link.
```
Comment (JV0812) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:22]:
```suggestion
This app supports logs and metrics for MongoDB instance. The MongoDB logs are generated in files as configured in this configuration file `/var/log/mongodb/mongodb.log`. For more details on MongoDB logs, refer to the [MongoDB documentation](https://docs.mongodb.com/manual/reference/log-messages/).
```

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 11]
```diff
@@ -29,15 +31,24 @@ Following are the [Fields](/docs/manage/fields/) which will be created as part o
...
    - 0 is the MongoDB's default log verbosity level, to include [Informational](https://docs.mongodb.com/manual/reference/log-messages/#std-label-log-severity-levels) messages.
    - 1 to 5 increases the verbosity level to include[ Debug](https://docs.mongodb.com/manual/reference/log-messages/#std-label-log-severity-levels) messages.
-2. Configure MongoDB to log to a Local file: Configuring MongoDB logs to go to log files. By default, MongoDB logs are stored in `/var/log/mongodb/mongodb.log`. The default directory for log files is listed in the MongoDB.conf file. To configure the log output destination to a log file, use one of the following settings, either in the [configuration file](https://docs.mongodb.com/manual/reference/configuration-options/) or command-line:
+2. You need to set the [profiling_level](https://www.mongodb.com/docs/manual/reference/method/db.setProfilingLevel/) to a value of 1. But default its set to 0. You can also configure value of [slowms](https://www.mongodb.com/docs/manual/reference/method/db.setProfilingLevel/#std-label-set-profiling-level-options-slowms) to a value greater then or equal to 100. Thus can be done using the below command in mongosh:
```
Comment (JV0812) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:49]:
```suggestion
2. You need to set the [profiling_level](https://www.mongodb.com/docs/manual/reference/method/db.setProfilingLevel/) to a value of 1. But by default it is set to 0. You can also configure the value of [slowms](https://www.mongodb.com/docs/manual/reference/method/db.setProfilingLevel/#std-label-set-profiling-level-options-slowms) to a value greater than or equal to 100. Thus can be done using the below command in mongosh:
```

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 12]
```diff
@@ -11,15 +11,17 @@ import TabItem from '@theme/TabItem';
...
 
 <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/MongoDB-OpenTelemetry/MongoDB-Schematics.png' alt="Schematics" />
 
-## Log types
+## Log and metrics types
 
-The MongoDB logs are generated in files as configured in the configuration file `/var/log/mongodb/mongodb.log`. For more details on MongoDB logs, see [this](https://docs.mongodb.com/manual/reference/log-messages/) link.
+This app supports metrics and logs for MongoDB instance. The MongoDB logs are generated in files as configured in the configuration file `/var/log/mongodb/mongodb.log`. For more details on MongoDB logs, see [this](https://docs.mongodb.com/manual/reference/log-messages/) link.
+
+The app supports metrics generated by the [MongoDB Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/MongoDBreceiver/documentation.md).
```
[the remark this answers is not in the document]
Comment (yasar-sumologic) [reply docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md]: fixed it.

File: docs/integrations/databases/opentelemetry/redis-opentelemetry.md [hunk 13]
```diff
@@ -114,7 +118,11 @@ import LogsIntro from '../../../reuse/apps/opentelemetry/send-logs-intro.md';
 <TabItem value="Linux">
 
 1. Copy the yaml at `/etc/otelcol-sumo/conf.d/` folder in the Redis instance that needs to be monitored.
-2. Restart the otelcol-sumo process using:
+2. Place Env file in the following directory:
```
Comment (JV0812) [inline docs/integrations/databases/opentelemetry/redis-opentelemetry.md:121]:
```suggestion
2. Move the env file in the following directory:
```

File: docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md [hunk 14]
```diff
@@ -252,29 +280,59 @@ The **MongoDB - Query Logs** dashboard shows read and write query trends.
...
 
 <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/MongoDB-OpenTelemetry/MongoDB-Query-Logs.png' alt="Query Logs" />
 
+### Sharding
+
+The **MongoDB - Sharding** dashboard shows sharding related errors, events, failures, and number of chunks moving between shards.
+
+Use this dashboard to:
+
+-   Identify sharding errors and warnings.
```
Comment (JV0812) [inline docs/integrations/databases/opentelemetry/mongodb-opentelemetry.md:294]:
```suggestion
- Identify sharding errors and warnings.
```

**Review by JV0812 — APPROVED** [review]
LGTM

**Review by jpipkin1 — APPROVED** [review]
Approved, @yasar-sumologic. I made minor changes (see commit "[Updates for review](https://github.com/SumoLogic/sumologic-documentation/pull/4517/commits/0be17a7c2ce196db628e93cbf5781cc5cf95a599)").

**Comment by jpipkin1** [comment]
@yasar-sumologic, is this PR ready to be published? 

If instead it needs to wait until a software release, add the **do-not-merge** label.

**Comment by jpipkin1** [comment]
@yasar-sumologic confirmed via Slack that this PR is ready to be published.

**Pull request merged**